### DISCUSSION TITLE
threading sanitizer warnings

### DISCUSSION THREAD

**Discussion by tonyhorrobin** [eProsima/Fast-DDS#5230] [General]
Dear all,

I compiled both fastdds 2.13.1 and the helloworld example with -fsanitizer=thread on Ubuntu 22.04.4 LTS using gcc-11.4.0 and got a list of warnings: double lock, lock order inversion and data races in libfastrtps.so and the main function.

Are these known issues and are they fixed in a later version of fastdds?

Cheers,

-Tony

**Comment by MiguelCompany** [comment]
We've seen false positives in the past with gcc-11, that's why we use gcc-12 on our thread sanitizer CI jobs

**Reply by tonyhorrobin** [reply]
Thanks.  Sorry I missed that one.
